run_unittests.py

```diff
@@ -9008,25 +9008,36 @@ def convert_args(argv):
         pytest_args += ['-k', ' or '.join(test_list)]
     return pytest_args
 
+def running_a_single_test(argv, cases):
```

Comment:
The name doesn't match what this function actually does: check if any arguments which looks like test names are present. Perhaps it should count them, and only avoid test parallelization if there is exactly one?

Great change to have, though!

Reply:
I split this out into https://github.com/mesonbuild/meson/pull/7724 and incorporated your suggestion :)